packages/protocol-utils/test/meta_transactions_v2_test.ts

```diff
@@ -0,0 +1,66 @@
+import { chaiSetup, } from '@0x/dev-utils';
+import { BigNumber } from '@0x/utils';
+import { expect } from 'chai';
+import { MetaTransactionV2 } from '../src/meta_transactions_v2';
+
+chaiSetup.configure();
+
+describe('mtxs v2', () => {
+    describe('MetaTransactionV2 (no fees)', () => {
+        const mtx = new MetaTransactionV2({
+            signer: '0x349e8d89e8b37214d9ce3949fc5754152c525bc3',
+            sender: '0x83c62b2e67dea0df2a27be0def7a22bd7102642c',
+            expirationTimeSeconds: new BigNumber(9101112),
+            salt: new BigNumber(2001),
+            callData: '0x12345678',
+            feeToken: '0xcc3c7ea403427154ec908203ba6c418bd699f7ce',
+            fees: [] as any,
+            chainId: 8008,
+            verifyingContract: '0x6701704d2421c64ee9aa93ec7f96ede81c4be77d',
+        });
+
+        it('can get the struct hash', () => {
+            const actual = mtx.getStructHash();
+            const expected = '0x57db4055edfed82a6d86103197c21390bf29412fb4585e08c708454e03d92516';
```

Comment:
Quick question: how are the expected hash generated? If it's computed via `ethers` or solidity then we are good. Just want to make sure the expected hash is computed in a different method so that we can make sure our method implementation is correct. Same for other hashes in the test

Reply:
It's computed via `ethers`, so hopefully we are fine

Reply:
Got it. Yeah should be fine 👍